In [0]:
%run ./utils/Forecasting_Utils

In [0]:
prod_sales_df = spark.sql("SELECT period_number, swire_period_date, state, key_brand, key_package, standard_physical_volume FROM stf_db.product_sales_series")
SeriesData.estimate_physical_volume_range(prod_sales_df)

# Short descriptor for cell contents
state_table_desc = lambda val,state: print(f"{val} in {state}")

# Key brand data model
kb_model = KeyBrandModel()
kb_model.quick_stats()

# Key package data model
kp_model = KeyPackageModel()
kp_model.quick_stats()

Average Physical Volume: 197048.62
Median Physical Volume: 182679.19
Standard Deviation: 46646.85

Physical Volume Range Expectation: 150401.76 to 243695.47

--- Key Brand Quick Stats ---
-----------------------------
• 29 unique key brand values
• 10 unique states
• 168 unique periods
   🠆 Min: Week ending on 11-15-2019
   🠆 Max: Week ending on 01-27-2023


--- Key Package Quick Stats ---
-------------------------------
• 27 unique key package values
• 10 unique states
• 168 unique periods
   🠆 Min: Week ending on 11-15-2019
   🠆 Max: Week ending on 01-27-2023




### Physical Volume per Period
Average|Median|Standard Deviation|
-------|------|------------------|
197048.62|182679.19|46646.85|

We will opt to recognize the range of 165K to 215K as **"Typical"** physical volume  

**Typical**: Physical volume within the range of 165K to 215K  
**Spike**: physical volume above the "typical" range  
**Dip**: Physical volume below the "typical" range  

### Tendencies   
* **Spikes** tend to occur about once a month between February and December
  * Likelihood and magnitude increases around major holidays
    * July 4, Labor Day, Thanksgiving, Christmas/New Year's
* **Dips** tend to occur in months following major holidays:
  * January, February, October
* Promotion prices may act as confounding factors on holiday spikes

### Top 10 Brands by Volume
Brand|Total Volume|
-----|------------|
Coke|17,816,471
Dr Pepper|7,372,236
Diet Dr Pepper|2,931,327
Cherry Coke|1,356,549
Dr Pepper Zero Sugar|561,346
Dr Pepper Cream Soda|555,059
Dr Pepper Cherry|386,120
Dr Pepper Cherry Zero Sugar|287,126
Dr Pepper Cream Soda Zero Sugar|256,115
Coke Cherry Vanilla|249,586

### Top 10 Package Types by Volume
Package Size|Package Count|Container|Total Volume|
------------|-------------|---------|------------|
12 Ounce|12-Pack|Can|20,970,705
.5 Liter|6-Pack|Bottle|2,792,235
2 Liter|Single|Bottle|2,673,159
12 Ounce|8-Pack|Pls Bottle|1,234,402
7.5 Ounce|10-Pack|Can|1,136,742
7.5 Ounce|6-Pack|Can|1,106,353
20 Ounce|Single|Bottle|1,036,213
12 Ounce|20-Pack|Can|933,024
12 Ounce|24-Pack|Can|521,157
1 Liter|Single|Bottle|160,260

### States Ranked by Volume
State|Total Volume|Volume % |
-----|------------|---------|
Arizona|12,888,928|38.9%|
Washington|8,298,735|25.1%|
Oregon|4,320,178|13.1%|
Colorado|3,793,326|11.5%|
Idaho|1,598,169|4.83%|
New Mexico|999,998|3.02%|
Nevada|801,919|2.42%|
Wyoming|224,999|0.68%|
Nebraska|118,136|0.36%|
California|59,775|0.18%|

In [0]:
display(prod_sales_df)

period_number swire_period_date state key_brand key_package standard_physical_volume 1 2019-11-15 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 2 2019-11-22 AZ CHERRY COKE .5L 6PK BOTTLE 2.25 3 2019-11-29 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 4 2019-12-06 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 5 2019-12-13 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 6 2019-12-20 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 7 2019-12-27 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 8 2020-01-03 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 9 2020-01-10 AZ CHERRY COKE .5L 6PK BOTTLE 2.75 10 2020-01-17 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 11 2020-01-24 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 12 2020-01-31 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 13 2020-02-07 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 14 2020-02-14 AZ CHERRY COKE .5L 6PK BOTTLE 2.25 15 2020-02-21 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 16 2020-02-28 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 17 2020-03-06 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 18 2020-03-13 AZ CHERRY COKE .5L 6PK BOTTLE 1.75 19 2020-03-20 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 20 2020-03-27 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 21 2020-04-03 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 22 2020-04-10 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 23 2020-04-17 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 24 2020-04-24 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 25 2020-05-01 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 26 2020-05-08 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 27 2020-05-15 AZ CHERRY COKE .5L 6PK BOTTLE 3.75 28 2020-05-22 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 29 2020-05-29 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 30 2020-06-05 AZ CHERRY COKE .5L 6PK BOTTLE 3.0 31 2020-06-12 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 32 2020-06-19 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 33 2020-06-26 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 34 2020-07-03 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 35 2020-07-10 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 36 2020-07-17 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 37 2020-07-24 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 38 2020-07-31 AZ CHERRY COKE .5L 6PK BOTTLE 2.25 39 2020-08-07 AZ CHERRY COKE .5L 6PK BOTTLE 1.75 40 2020-08-14 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 41 2020-08-21 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 42 2020-08-28 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 43 2020-09-04 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 44 2020-09-11 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 45 2020-09-18 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 46 2020-09-25 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 47 2020-10-02 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 48 2020-10-09 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 49 2020-10-16 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 50 2020-10-23 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 51 2020-10-30 AZ CHERRY COKE .5L 6PK BOTTLE 3.25 52 2020-11-06 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 53 2020-11-13 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 54 2020-11-20 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 55 2020-12-04 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 56 2020-12-11 AZ CHERRY COKE .5L 6PK BOTTLE 1.75 57 2020-12-18 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 58 2020-12-25 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 59 2021-01-01 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 60 2021-01-08 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 61 2021-01-15 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 62 2021-01-22 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 70 2021-03-19 AZ CHERRY COKE .5L 6PK BOTTLE 4.0 71 2021-03-26 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 72 2021-04-02 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 73 2021-04-09 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 75 2021-04-23 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 76 2021-04-30 AZ CHERRY COKE .5L 6PK BOTTLE 2.75 77 2021-05-07 AZ CHERRY COKE .5L 6PK BOTTLE 3.5 78 2021-05-14 AZ CHERRY COKE .5L 6PK BOTTLE 1.25 79 2021-05-21 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 80 2021-05-28 AZ CHERRY COKE .5L 6PK BOTTLE 1.25 81 2021-06-04 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 82 2021-06-11 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 83 2021-06-18 AZ CHERRY COKE .5L 6PK BOTTLE 1.25 84 2021-06-25 AZ CHERRY COKE .5L 6PK BOTTLE 3.25 85 2021-07-02 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 86 2021-07-09 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 87 2021-07-16 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 88 2021-07-23 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 89 2021-07-30 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 90

Output can only be rendered in Databricks

Output can only be rendered in Databricks

Output can only be rendered in Databricks

Output can only be rendered in Databricks

Output can only be rendered in Databricks

While the proportion ranges exhibit some non-trivial variation, the "rank" of the brands in each state largely agrees with the overall aggregate above rank 4. Variation gets significantly more interesting begining with rank 5. For instance:  
* While Mexican Coke ranks 5th (4.12%) in Nevada, it ranks outside of the top 10 (0.69%) in New Mexico.
* While Dr Pepper Zero Sugar ranks 5th overall, it ranks outside of the top 5 in 8 of 10 states. 
  * It's overall ranking is largely attributed to the 5th place ranking in Arizona, which is the largest market, by far.

### Top Brands by Percentage of Volume
Overall Rank|Brand|Min Volume %|Max Volume %|
------------|-----|------------|------------|
1|Coke|40.2% (CA)|58% (AZ)|
2|Dr Pepper|19.4% (WA)|33.8% (NE)|
3|Diet Dr Pepper|7.41% (AZ)|12.4% (CA)|
4|Cherry Coke|2.63% (NM)|5.59% (WA)|
5|Dr Pepper Zero Sugar|0.61% (CA)|2.10% (ID)|
6|Dr Pepper Cream Soda|1.23% (AZ)|4.21% (CA)|
7|Dr Pepper Cherry|1.05% (WA)|2.99% (CA)|
8|Dr Pepper Cherry Zero Sugar|0.27% (CA)|1.17% (NM)|
9|Dr Pepper Cream Soda Zero Sugar|0.25% (CA)|1.48% (ID)|
10|Coke Cherry Vanilla|0.43% (CA)|1.09% (CA,WA)|

In [0]:
display(prod_sales_df)

period_number swire_period_date state key_brand key_package standard_physical_volume 1 2019-11-15 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 2 2019-11-22 AZ CHERRY COKE .5L 6PK BOTTLE 2.25 3 2019-11-29 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 4 2019-12-06 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 5 2019-12-13 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 6 2019-12-20 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 7 2019-12-27 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 8 2020-01-03 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 9 2020-01-10 AZ CHERRY COKE .5L 6PK BOTTLE 2.75 10 2020-01-17 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 11 2020-01-24 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 12 2020-01-31 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 13 2020-02-07 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 14 2020-02-14 AZ CHERRY COKE .5L 6PK BOTTLE 2.25 15 2020-02-21 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 16 2020-02-28 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 17 2020-03-06 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 18 2020-03-13 AZ CHERRY COKE .5L 6PK BOTTLE 1.75 19 2020-03-20 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 20 2020-03-27 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 21 2020-04-03 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 22 2020-04-10 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 23 2020-04-17 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 24 2020-04-24 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 25 2020-05-01 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 26 2020-05-08 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 27 2020-05-15 AZ CHERRY COKE .5L 6PK BOTTLE 3.75 28 2020-05-22 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 29 2020-05-29 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 30 2020-06-05 AZ CHERRY COKE .5L 6PK BOTTLE 3.0 31 2020-06-12 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 32 2020-06-19 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 33 2020-06-26 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 34 2020-07-03 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 35 2020-07-10 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 36 2020-07-17 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 37 2020-07-24 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 38 2020-07-31 AZ CHERRY COKE .5L 6PK BOTTLE 2.25 39 2020-08-07 AZ CHERRY COKE .5L 6PK BOTTLE 1.75 40 2020-08-14 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 41 2020-08-21 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 42 2020-08-28 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 43 2020-09-04 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 44 2020-09-11 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 45 2020-09-18 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 46 2020-09-25 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 47 2020-10-02 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 48 2020-10-09 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 49 2020-10-16 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 50 2020-10-23 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 51 2020-10-30 AZ CHERRY COKE .5L 6PK BOTTLE 3.25 52 2020-11-06 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 53 2020-11-13 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 54 2020-11-20 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 55 2020-12-04 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 56 2020-12-11 AZ CHERRY COKE .5L 6PK BOTTLE 1.75 57 2020-12-18 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 58 2020-12-25 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 59 2021-01-01 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 60 2021-01-08 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 61 2021-01-15 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 62 2021-01-22 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 70 2021-03-19 AZ CHERRY COKE .5L 6PK BOTTLE 4.0 71 2021-03-26 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 72 2021-04-02 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 73 2021-04-09 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 75 2021-04-23 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 76 2021-04-30 AZ CHERRY COKE .5L 6PK BOTTLE 2.75 77 2021-05-07 AZ CHERRY COKE .5L 6PK BOTTLE 3.5 78 2021-05-14 AZ CHERRY COKE .5L 6PK BOTTLE 1.25 79 2021-05-21 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 80 2021-05-28 AZ CHERRY COKE .5L 6PK BOTTLE 1.25 81 2021-06-04 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 82 2021-06-11 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 83 2021-06-18 AZ CHERRY COKE .5L 6PK BOTTLE 1.25 84 2021-06-25 AZ CHERRY COKE .5L 6PK BOTTLE 3.25 85 2021-07-02 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 86 2021-07-09 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 87 2021-07-16 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 88 2021-07-23 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 89 2021-07-30 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 90

Output can only be rendered in Databricks

Output can only be rendered in Databricks

The overwhelming majority of units are the 12OZ 12-Pack Can packaging. The .5L 6-Pack Bottle and the 2L Single Bottle are typically next in order. NE and CA are the two states that deviate from this, while still keeping those package types in their top 5.


### Top Packages Types by Percentage of Volume
Overall Rank|Package Size|Package Count|Container|Min Volume %|Max Volume %|
------------|------------|-------------|---------|------------|------------|
1|12 Ounce|12-Pack|Can|57% (NV)|73.4% (CA)|
2|.5 Liter|6-Pack|Bottle|4.84% (CA)|15.6% (NE)|
3|2 Liter|Single|Bottle|3.61% (CA)|11.19% (AZ)|
4|12 Ounce|8-Pack|Pls Bottle|2.84% (AZ)|5.50% (WY)|
5|7.5 Ounce|10-Pack|Can|1.46% (NE)|4.96% (NV)|
6|7.5 Ounce|6-Pack|Can|2.45% (NE)|4.40% (CO)|
7|20 Ounce|Single|Bottle|2.46% (AZ)|4.47% (CA)|
8|12 Ounce|20-Pack|Can|0.88% (OR)|4.80% (AZ)|
9|12 Ounce|24-Pack|Can|0.0% (NE)|2.88% (WA)|
10|1 Liter|Single|Bottle|0.0% (NE)|1.35% (WA)|

In [0]:
display(prod_sales_df)

period_number swire_period_date state key_brand key_package standard_physical_volume 1 2019-11-15 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 2 2019-11-22 AZ CHERRY COKE .5L 6PK BOTTLE 2.25 3 2019-11-29 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 4 2019-12-06 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 5 2019-12-13 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 6 2019-12-20 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 7 2019-12-27 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 8 2020-01-03 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 9 2020-01-10 AZ CHERRY COKE .5L 6PK BOTTLE 2.75 10 2020-01-17 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 11 2020-01-24 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 12 2020-01-31 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 13 2020-02-07 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 14 2020-02-14 AZ CHERRY COKE .5L 6PK BOTTLE 2.25 15 2020-02-21 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 16 2020-02-28 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 17 2020-03-06 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 18 2020-03-13 AZ CHERRY COKE .5L 6PK BOTTLE 1.75 19 2020-03-20 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 20 2020-03-27 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 21 2020-04-03 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 22 2020-04-10 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 23 2020-04-17 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 24 2020-04-24 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 25 2020-05-01 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 26 2020-05-08 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 27 2020-05-15 AZ CHERRY COKE .5L 6PK BOTTLE 3.75 28 2020-05-22 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 29 2020-05-29 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 30 2020-06-05 AZ CHERRY COKE .5L 6PK BOTTLE 3.0 31 2020-06-12 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 32 2020-06-19 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 33 2020-06-26 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 34 2020-07-03 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 35 2020-07-10 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 36 2020-07-17 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 37 2020-07-24 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 38 2020-07-31 AZ CHERRY COKE .5L 6PK BOTTLE 2.25 39 2020-08-07 AZ CHERRY COKE .5L 6PK BOTTLE 1.75 40 2020-08-14 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 41 2020-08-21 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 42 2020-08-28 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 43 2020-09-04 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 44 2020-09-11 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 45 2020-09-18 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 46 2020-09-25 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 47 2020-10-02 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 48 2020-10-09 AZ CHERRY COKE .5L 6PK BOTTLE 1.5 49 2020-10-16 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 50 2020-10-23 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 51 2020-10-30 AZ CHERRY COKE .5L 6PK BOTTLE 3.25 52 2020-11-06 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 53 2020-11-13 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 54 2020-11-20 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 55 2020-12-04 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 56 2020-12-11 AZ CHERRY COKE .5L 6PK BOTTLE 1.75 57 2020-12-18 AZ CHERRY COKE .5L 6PK BOTTLE 2.0 58 2020-12-25 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 59 2021-01-01 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 60 2021-01-08 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 61 2021-01-15 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 62 2021-01-22 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 70 2021-03-19 AZ CHERRY COKE .5L 6PK BOTTLE 4.0 71 2021-03-26 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 72 2021-04-02 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 73 2021-04-09 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 75 2021-04-23 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 76 2021-04-30 AZ CHERRY COKE .5L 6PK BOTTLE 2.75 77 2021-05-07 AZ CHERRY COKE .5L 6PK BOTTLE 3.5 78 2021-05-14 AZ CHERRY COKE .5L 6PK BOTTLE 1.25 79 2021-05-21 AZ CHERRY COKE .5L 6PK BOTTLE 0.0 80 2021-05-28 AZ CHERRY COKE .5L 6PK BOTTLE 1.25 81 2021-06-04 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 82 2021-06-11 AZ CHERRY COKE .5L 6PK BOTTLE 0.25 83 2021-06-18 AZ CHERRY COKE .5L 6PK BOTTLE 1.25 84 2021-06-25 AZ CHERRY COKE .5L 6PK BOTTLE 3.25 85 2021-07-02 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 86 2021-07-09 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 87 2021-07-16 AZ CHERRY COKE .5L 6PK BOTTLE 0.75 88 2021-07-23 AZ CHERRY COKE .5L 6PK BOTTLE 1.0 89 2021-07-30 AZ CHERRY COKE .5L 6PK BOTTLE 0.5 90

Output can only be rendered in Databricks

Output can only be rendered in Databricks

## State + Key Brand

#### -- Physical Volume + Promotion --  
More promotional activity does tend to elevate volume. However, prolonged promotions seem to affect different markets differently.

Provided circumstances surrounding $0 promotions, there is no way to be certain what effects these specific events tend to generate, if any.  
In the observed instances, these events tend to occur:  
* During prolonged dips (i.e.: January)
* Within one or two periods before and after spikes in promotional activity


#### -- Promotion Comparison --  
* The three Dollar Promotion values per period, which are each computed on different average standard price windows, mostly agree.
  * To the extent that they differ, smaller price windows tend to have higher promotion values

In [0]:
b_state = kb_model.states_list[0]
brand = kb_model.keys_list[3]
state_table_desc(brand, b_state)

display(kb_model.get_filtered_series(b_state, brand))

COKE in AZ


state key_brand period_number swire_period_date avg_standard_physical_volume dollar_promo_6wk dollar_promo_8wk dollar_promo_12wk AZ COKE 1 2019-11-15 20.58 79.06 79.06 79.06 AZ COKE 2 2019-11-22 14.25 27.06 27.06 27.06 AZ COKE 3 2019-11-29 19.79 38.09 38.09 38.09 AZ COKE 4 2019-12-06 15.31 24.74 24.74 24.74 AZ COKE 5 2019-12-13 15.05 24.84 24.84 24.84 AZ COKE 6 2019-12-20 16.69 26.92 26.92 26.92 AZ COKE 7 2019-12-27 26.0 12.62 12.62 11.92 AZ COKE 8 2020-01-03 15.8 44.55 43.93 43.93 AZ COKE 9 2020-01-10 10.95 22.65 22.65 22.25 AZ COKE 10 2020-01-17 13.13 14.19 14.19 13.72 AZ COKE 11 2020-01-24 14.86 3.91 3.48 3.04 AZ COKE 12 2020-01-31 12.8 33.12 32.65 31.72 AZ COKE 13 2020-02-07 18.19 41.55 41.01 39.93 AZ COKE 14 2020-02-14 14.07 36.27 35.27 34.26 AZ COKE 15 2020-02-21 15.44 21.21 20.73 19.77 AZ COKE 16 2020-02-28 12.16 27.81 27.81 26.51 AZ COKE 17 2020-03-06 14.73 39.24 39.24 38.28 AZ COKE 18 2020-03-13 17.07 37.6 37.02 36.45 AZ COKE 19 2020-03-20 23.98 23.53 23.53 22.14 AZ COKE 20 2020-03-27 18.28 51.28 51.28 50.63 AZ COKE 21 2020-04-03 16.09 21.78 21.31 21.31 AZ COKE 22 2020-04-10 16.47 31.46 30.86 30.27 AZ COKE 23 2020-04-17 23.28 25.35 25.35 24.7 AZ COKE 24 2020-04-24 18.02 31.02 31.02 30.46 AZ COKE 25 2020-05-01 21.61 29.4 29.4 28.78 AZ COKE 26 2020-05-08 17.62 54.82 54.82 54.82 AZ COKE 27 2020-05-15 19.23 38.04 38.04 37.38 AZ COKE 28 2020-05-22 18.73 43.51 43.51 43.51 AZ COKE 29 2020-05-29 24.57 43.34 43.34 43.34 AZ COKE 30 2020-06-05 16.43 57.16 57.16 57.16 AZ COKE 31 2020-06-12 20.33 46.14 46.14 46.14 AZ COKE 32 2020-06-19 19.05 71.17 71.17 71.78 AZ COKE 33 2020-06-26 18.95 36.54 36.54 36.54 AZ COKE 34 2020-07-03 14.92 62.42 62.42 62.42 AZ COKE 35 2020-07-10 27.22 44.18 44.18 44.18 AZ COKE 36 2020-07-17 16.97 59.1 58.49 59.1 AZ COKE 37 2020-07-24 16.21 40.08 40.08 40.08 AZ COKE 38 2020-07-31 15.3 90.48 90.48 90.48 AZ COKE 39 2020-08-07 18.7 63.99 63.99 63.99 AZ COKE 40 2020-08-14 17.25 40.52 40.52 40.52 AZ COKE 41 2020-08-21 13.8 50.72 50.72 50.72 AZ COKE 42 2020-08-28 18.97 13.29 13.29 13.29 AZ COKE 43 2020-09-04 15.81 93.34 93.34 93.34 AZ COKE 44 2020-09-11 26.35 32.99 32.99 32.99 AZ COKE 45 2020-09-18 14.59 40.98 40.41 40.41 AZ COKE 46 2020-09-25 18.7 79.6 78.87 78.87 AZ COKE 47 2020-10-02 18.23 59.18 58.58 57.97 AZ COKE 48 2020-10-09 18.56 37.88 36.67 35.47 AZ COKE 49 2020-10-16 20.01 69.94 69.24 67.14 AZ COKE 50 2020-10-23 13.29 74.97 73.75 71.92 AZ COKE 51 2020-10-30 13.57 76.16 75.6 73.93 AZ COKE 52 2020-11-06 18.51 79.46 78.68 76.34 AZ COKE 53 2020-11-13 15.33 103.28 103.28 101.94 AZ COKE 54 2020-11-20 14.59 63.84 63.84 63.24 AZ COKE 55 2020-12-04 15.1 9.7 9.7 9.13 AZ COKE 56 2020-12-11 13.83 49.31 49.31 49.31 AZ COKE 57 2020-12-18 15.64 58.69 59.34 59.34 AZ COKE 58 2020-12-25 20.86 118.34 118.34 119.27 AZ COKE 59 2021-01-01 18.22 85.43 86.18 86.93 AZ COKE 60 2021-01-08 15.89 69.96 69.96 71.28 AZ COKE 61 2021-01-15 16.58 28.91 28.91 29.56 AZ COKE 62 2021-01-22 19.81 0.0 0.0 0.0 AZ COKE 63 2021-01-29 13.15 31.09 31.09 31.09 AZ COKE 64 2021-02-05 15.86 60.5 59.84 59.84 AZ COKE 65 2021-02-12 24.83 20.08 20.08 19.34 AZ COKE 66 2021-02-19 15.01 51.56 50.95 50.33 AZ COKE 67 2021-02-26 16.09 48.18 47.5 46.82 AZ COKE 68 2021-03-05 15.03 78.32 78.32 77.03 AZ COKE 69 2021-03-12 14.42 34.82 34.82 34.22 AZ COKE 70 2021-03-19 16.22 59.3 59.3 59.3 AZ COKE 71 2021-03-26 15.85 59.42 59.42 58.78 AZ COKE 72 2021-04-02 13.93 37.23 37.23 37.23 AZ COKE 73 2021-04-09 29.62 16.1 16.1 16.95 AZ COKE 74 2021-04-16 15.27 58.52 58.52 58.52 AZ COKE 75 2021-04-23 13.19 50.15 50.15 50.72 AZ COKE 76 2021-04-30 17.51 82.43 82.43 83.2 AZ COKE 77 2021-05-07 16.63 68.74 68.74 69.4 AZ COKE 78 2021-05-14 16.63 30.22 30.22 30.22 AZ COKE 79 2021-05-21 15.61 19.1 19.1 19.72 AZ COKE 80 2021-05-28 19.61 68.48 68.48 68.48 AZ COKE 81 2021-06-04 28.12 54.81 54.81 55.65 AZ COKE 82 2021-06-11 17.63 12.62 12.62 13.22 AZ COKE 83 2021-06-18 17.27 40.99 41.66 41.66 AZ COKE 84 2021-06-25 17.53 40.58 41.25 41.25 AZ COKE 85 2021-07-02 14.53 88.61 89.23 89.86 AZ CO

Output can only be rendered in Databricks

Output can only be rendered in Databricks

In [0]:
b_state = kb_model.states_list[0]
brand = kb_model.keys_list[14]
state_table_desc(brand, b_state)

display(kb_model.get_filtered_series(b_state, brand))

DR PEPPER in AZ


state key_brand period_number swire_period_date avg_standard_physical_volume dollar_promo_6wk dollar_promo_8wk dollar_promo_12wk AZ DR PEPPER 1 2019-11-15 6.93 27.93 27.93 27.93 AZ DR PEPPER 2 2019-11-22 5.16 15.31 15.31 15.31 AZ DR PEPPER 3 2019-11-29 6.62 17.17 17.17 17.17 AZ DR PEPPER 4 2019-12-06 5.2 10.3 10.3 10.3 AZ DR PEPPER 5 2019-12-13 4.78 12.08 12.08 12.08 AZ DR PEPPER 6 2019-12-20 5.53 14.61 14.61 14.61 AZ DR PEPPER 7 2019-12-27 8.49 8.13 8.13 8.13 AZ DR PEPPER 8 2020-01-03 5.21 19.03 18.81 18.81 AZ DR PEPPER 9 2020-01-10 4.15 13.66 13.49 13.49 AZ DR PEPPER 10 2020-01-17 4.87 8.22 8.03 8.03 AZ DR PEPPER 11 2020-01-24 5.07 3.82 3.66 3.49 AZ DR PEPPER 12 2020-01-31 4.44 15.03 14.85 14.66 AZ DR PEPPER 13 2020-02-07 6.57 16.55 16.33 15.9 AZ DR PEPPER 14 2020-02-14 5.18 19.77 19.35 18.93 AZ DR PEPPER 15 2020-02-21 5.79 7.11 6.92 6.53 AZ DR PEPPER 16 2020-02-28 4.54 13.15 12.97 12.44 AZ DR PEPPER 17 2020-03-06 4.95 11.61 11.43 11.06 AZ DR PEPPER 18 2020-03-13 5.93 14.02 13.8 13.36 AZ DR PEPPER 19 2020-03-20 9.31 12.54 12.54 11.96 AZ DR PEPPER 20 2020-03-27 7.73 21.82 21.54 21.27 AZ DR PEPPER 21 2020-04-03 6.27 9.38 9.38 9.18 AZ DR PEPPER 22 2020-04-10 6.77 13.03 13.03 12.77 AZ DR PEPPER 23 2020-04-17 9.36 22.17 22.17 21.9 AZ DR PEPPER 24 2020-04-24 6.72 12.18 12.18 11.96 AZ DR PEPPER 25 2020-05-01 8.17 12.73 12.73 12.47 AZ DR PEPPER 26 2020-05-08 6.99 25.71 25.71 25.44 AZ DR PEPPER 27 2020-05-15 7.63 21.23 21.23 21.23 AZ DR PEPPER 28 2020-05-22 6.84 15.24 15.24 15.24 AZ DR PEPPER 29 2020-05-29 9.48 19.33 19.33 19.05 AZ DR PEPPER 30 2020-06-05 6.31 21.98 21.98 21.98 AZ DR PEPPER 31 2020-06-12 8.02 19.94 19.94 19.69 AZ DR PEPPER 32 2020-06-19 7.69 28.42 28.68 28.42 AZ DR PEPPER 33 2020-06-26 7.67 15.82 15.82 15.82 AZ DR PEPPER 34 2020-07-03 5.78 24.94 24.94 24.94 AZ DR PEPPER 35 2020-07-10 11.2 20.62 20.3 20.62 AZ DR PEPPER 36 2020-07-17 7.17 29.26 29.26 29.26 AZ DR PEPPER 37 2020-07-24 6.58 20.53 20.53 20.53 AZ DR PEPPER 38 2020-07-31 6.29 38.57 38.57 38.31 AZ DR PEPPER 39 2020-08-07 7.66 27.5 27.5 27.5 AZ DR PEPPER 40 2020-08-14 7.39 16.4 16.4 16.4 AZ DR PEPPER 41 2020-08-21 6.24 20.16 20.41 20.41 AZ DR PEPPER 42 2020-08-28 8.02 8.77 8.77 8.77 AZ DR PEPPER 43 2020-09-04 6.01 35.68 35.94 35.94 AZ DR PEPPER 44 2020-09-11 9.48 22.0 22.0 22.0 AZ DR PEPPER 45 2020-09-18 5.64 24.75 24.75 24.5 AZ DR PEPPER 46 2020-09-25 6.58 31.08 30.79 30.79 AZ DR PEPPER 47 2020-10-02 7.48 29.7 29.43 29.43 AZ DR PEPPER 48 2020-10-09 7.39 20.15 19.89 19.37 AZ DR PEPPER 49 2020-10-16 7.68 34.46 33.89 33.6 AZ DR PEPPER 50 2020-10-23 5.27 38.99 38.73 38.21 AZ DR PEPPER 51 2020-10-30 5.63 39.97 39.97 39.47 AZ DR PEPPER 52 2020-11-06 7.16 38.84 38.84 37.87 AZ DR PEPPER 53 2020-11-13 6.8 54.45 54.45 54.14 AZ DR PEPPER 54 2020-11-20 5.94 26.97 26.97 26.97 AZ DR PEPPER 55 2020-12-04 7.05 7.71 7.71 7.71 AZ DR PEPPER 56 2020-12-11 6.13 23.32 23.62 23.62 AZ DR PEPPER 57 2020-12-18 6.72 23.3 23.3 23.59 AZ DR PEPPER 58 2020-12-25 8.67 49.79 50.18 50.18 AZ DR PEPPER 59 2021-01-01 8.15 38.79 39.13 39.13 AZ DR PEPPER 60 2021-01-08 6.89 25.81 25.81 26.1 AZ DR PEPPER 61 2021-01-15 7.53 10.56 10.87 10.87 AZ DR PEPPER 62 2021-01-22 6.53 0.0 0.0 0.0 AZ DR PEPPER 63 2021-01-29 5.66 21.67 21.38 21.38 AZ DR PEPPER 64 2021-02-05 6.64 30.29 29.99 29.99 AZ DR PEPPER 65 2021-02-12 9.36 10.06 9.74 9.41 AZ DR PEPPER 66 2021-02-19 6.51 21.33 21.04 20.75 AZ DR PEPPER 67 2021-02-26 6.91 23.18 22.86 22.54 AZ DR PEPPER 68 2021-03-05 6.99 30.49 30.49 29.89 AZ DR PEPPER 69 2021-03-12 7.06 16.0 15.71 15.42 AZ DR PEPPER 70 2021-03-19 6.95 22.05 22.05 21.76 AZ DR PEPPER 71 2021-03-26 6.84 28.49 28.49 28.2 AZ DR PEPPER 72 2021-04-02 5.86 16.8 16.8 16.54 AZ DR PEPPER 73 2021-04-09 12.64 13.73 13.73 13.35 AZ DR PEPPER 74 2021-04-16 6.74 27.41 27.41 27.41 AZ DR PEPPER 75 2021-04-23 5.67 21.88 22.13 22.13 AZ DR PEPPER 76 2021-04-30 7.05 33.39 33.39 33.39 AZ DR PEPPER 77 2021-05-07 7.17 29.73 29.73 30.03 AZ DR PEPPER 78 2021-05-14 7.22 18.28 18.59 18.59 AZ DR PEPPER 79 2021-05-21 6.59 9.75

Output can only be rendered in Databricks

Output can only be rendered in Databricks

In [0]:
b_state = kb_model.states_list[2]
brand = kb_model.keys_list[3]
state_table_desc(brand, b_state)

display(kb_model.get_filtered_series(b_state, brand))

COKE in CO


state key_brand period_number swire_period_date avg_standard_physical_volume dollar_promo_6wk dollar_promo_8wk dollar_promo_12wk CO COKE 1 2019-11-15 5.91 0.0 0.0 0.0 CO COKE 2 2019-11-22 8.65 0.0 0.0 0.0 CO COKE 3 2019-11-29 17.38 20.07 20.07 20.07 CO COKE 4 2019-12-06 9.31 0.0 0.0 0.0 CO COKE 5 2019-12-13 7.57 0.0 0.0 0.0 CO COKE 6 2019-12-20 15.55 47.67 47.67 47.67 CO COKE 7 2019-12-27 17.73 53.51 53.51 53.51 CO COKE 8 2020-01-03 11.41 25.79 25.79 25.41 CO COKE 9 2020-01-10 5.58 0.0 0.0 0.0 CO COKE 10 2020-01-17 7.2 0.0 0.0 0.0 CO COKE 11 2020-01-24 12.88 21.18 21.18 20.81 CO COKE 12 2020-01-31 5.75 0.0 0.0 0.0 CO COKE 13 2020-02-07 15.03 57.35 56.92 56.49 CO COKE 14 2020-02-14 6.28 40.87 40.61 40.08 CO COKE 15 2020-02-21 9.56 21.15 20.53 19.91 CO COKE 16 2020-02-28 6.2 28.77 28.52 28.01 CO COKE 17 2020-03-06 8.71 18.15 17.85 17.26 CO COKE 18 2020-03-13 7.47 44.92 44.92 44.28 CO COKE 19 2020-03-20 18.2 22.44 22.44 21.92 CO COKE 20 2020-03-27 12.74 1.54 1.54 1.15 CO COKE 21 2020-04-03 8.38 35.92 35.92 35.92 CO COKE 22 2020-04-10 10.37 0.0 0.0 0.0 CO COKE 23 2020-04-17 13.56 15.88 15.88 15.88 CO COKE 24 2020-04-24 8.14 16.49 16.76 16.76 CO COKE 25 2020-05-01 9.87 4.08 4.08 4.08 CO COKE 26 2020-05-08 10.63 0.0 0.0 0.34 CO COKE 27 2020-05-15 9.45 0.0 0.0 0.0 CO COKE 28 2020-05-22 11.9 27.45 27.45 27.45 CO COKE 29 2020-05-29 17.98 51.52 52.04 52.04 CO COKE 30 2020-06-05 7.77 0.0 0.0 0.0 CO COKE 31 2020-06-12 8.56 26.69 26.69 27.0 CO COKE 32 2020-06-19 15.97 7.31 7.31 7.77 CO COKE 33 2020-06-26 10.1 10.92 10.92 11.27 CO COKE 34 2020-07-03 8.44 25.77 26.08 26.39 CO COKE 35 2020-07-10 16.99 42.01 42.53 43.04 CO COKE 36 2020-07-17 9.48 15.58 15.58 15.92 CO COKE 37 2020-07-24 13.05 5.42 5.84 6.26 CO COKE 38 2020-07-31 12.1 13.15 13.15 13.53 CO COKE 39 2020-08-07 16.39 3.29 3.29 3.76 CO COKE 40 2020-08-14 10.62 22.59 22.59 22.98 CO COKE 41 2020-08-21 9.82 6.06 6.06 6.06 CO COKE 42 2020-08-28 10.38 0.0 0.0 0.0 CO COKE 43 2020-09-04 7.5 11.4 11.4 11.4 CO COKE 44 2020-09-11 16.68 18.49 17.99 17.99 CO COKE 45 2020-09-18 6.27 0.0 0.0 0.0 CO COKE 46 2020-09-25 8.35 11.94 11.6 11.26 CO COKE 47 2020-10-02 7.64 13.09 12.77 12.13 CO COKE 48 2020-10-09 17.22 0.0 0.0 0.0 CO COKE 49 2020-10-16 14.22 11.21 10.28 9.34 CO COKE 50 2020-10-23 7.33 52.35 52.35 51.71 CO COKE 51 2020-10-30 8.48 21.09 21.09 20.4 CO COKE 52 2020-11-06 10.22 52.72 52.72 51.85 CO COKE 53 2020-11-13 7.96 50.87 50.87 50.54 CO COKE 54 2020-11-20 9.54 60.41 60.41 60.41 CO COKE 55 2020-12-04 11.68 0.0 0.0 0.0 CO COKE 56 2020-12-11 7.55 50.91 50.91 50.91 CO COKE 57 2020-12-18 10.43 67.38 67.38 67.38 CO COKE 58 2020-12-25 9.24 46.12 46.12 46.12 CO COKE 59 2021-01-01 9.71 39.21 39.21 39.21 CO COKE 60 2021-01-08 13.04 93.05 93.05 93.05 CO COKE 61 2021-01-15 7.74 54.37 54.37 54.37 CO COKE 62 2021-01-22 10.75 49.37 49.37 49.37 CO COKE 63 2021-01-29 6.86 5.3 5.02 5.02 CO COKE 64 2021-02-05 8.89 22.49 22.49 22.49 CO COKE 65 2021-02-12 12.56 42.72 42.72 42.72 CO COKE 66 2021-02-19 8.7 51.58 51.58 51.58 CO COKE 67 2021-02-26 8.18 73.44 73.44 73.44 CO COKE 68 2021-03-05 8.65 0.0 0.0 0.0 CO COKE 69 2021-03-12 9.79 14.08 14.08 13.67 CO COKE 70 2021-03-19 12.23 3.31 2.84 2.37 CO COKE 71 2021-03-26 9.81 8.87 8.47 7.66 CO COKE 72 2021-04-02 8.05 0.0 0.0 0.0 CO COKE 73 2021-04-09 13.49 0.0 0.0 0.0 CO COKE 74 2021-04-16 8.04 3.84 3.84 3.2 CO COKE 75 2021-04-23 8.91 0.0 0.0 0.0 CO COKE 76 2021-04-30 8.59 0.0 0.0 0.0 CO COKE 77 2021-05-07 8.18 21.9 21.9 21.9 CO COKE 78 2021-05-14 9.46 3.96 3.96 4.32 CO COKE 79 2021-05-21 8.71 3.01 3.35 4.02 CO COKE 80 2021-05-28 10.57 0.0 0.0 0.0 CO COKE 81 2021-06-04 18.17 30.23 31.35 31.91 CO COKE 82 2021-06-11 9.5 0.0 0.0 0.0 CO COKE 83 2021-06-18 10.91 25.47 25.47 26.36 CO COKE 84 2021-06-25 15.54 0.0 0.0 0.0 CO COKE 85 2021-07-02 10.54 0.0 0.0 0.0 CO COKE 86 2021-07-09 20.62 31.0 30.4 29.79 CO COKE 87 2021-07-16 9.28 14.58 14.22 13.85 CO COKE 88 2021-07-23 11.11 0.0 0.0 0.0 CO COKE 89 2021-07-30 11.66 3.02 2.59 1.72 CO COKE 90 2021-08-06 12.93 0.0 0.0 0.0 CO COK

Output can only be rendered in Databricks

Output can only be rendered in Databricks

In [0]:
b_state = kb_model.states_list[2]
brand = kb_model.keys_list[14]
state_table_desc(brand, b_state)

display(kb_model.get_filtered_series(b_state, brand))

DR PEPPER in CO


state key_brand period_number swire_period_date avg_standard_physical_volume dollar_promo_6wk dollar_promo_8wk dollar_promo_12wk CO DR PEPPER 1 2019-11-15 4.84 8.98 8.98 8.98 CO DR PEPPER 2 2019-11-22 5.24 0.0 0.0 0.0 CO DR PEPPER 3 2019-11-29 10.01 16.62 16.62 16.62 CO DR PEPPER 4 2019-12-06 5.63 1.22 1.22 1.22 CO DR PEPPER 5 2019-12-13 4.86 0.0 0.0 0.0 CO DR PEPPER 6 2019-12-20 8.54 20.39 20.39 20.39 CO DR PEPPER 7 2019-12-27 10.2 40.4 40.4 40.4 CO DR PEPPER 8 2020-01-03 4.54 5.01 5.01 5.01 CO DR PEPPER 9 2020-01-10 4.42 7.19 7.19 7.02 CO DR PEPPER 10 2020-01-17 4.93 7.4 7.21 7.21 CO DR PEPPER 11 2020-01-24 4.37 16.16 15.98 15.98 CO DR PEPPER 12 2020-01-31 4.89 18.07 17.88 17.69 CO DR PEPPER 13 2020-02-07 9.72 41.95 41.65 41.35 CO DR PEPPER 14 2020-02-14 3.91 23.25 23.07 22.9 CO DR PEPPER 15 2020-02-21 3.89 19.39 19.22 19.04 CO DR PEPPER 16 2020-02-28 4.62 17.77 17.77 17.39 CO DR PEPPER 17 2020-03-06 4.45 14.47 14.29 14.11 CO DR PEPPER 18 2020-03-13 4.6 26.21 26.42 26.01 CO DR PEPPER 19 2020-03-20 7.32 0.0 0.0 0.0 CO DR PEPPER 20 2020-03-27 6.69 13.65 13.65 13.65 CO DR PEPPER 21 2020-04-03 5.61 14.21 14.21 14.21 CO DR PEPPER 22 2020-04-10 5.69 3.52 3.52 3.72 CO DR PEPPER 23 2020-04-17 8.35 13.06 13.06 13.06 CO DR PEPPER 24 2020-04-24 5.68 8.49 8.49 8.49 CO DR PEPPER 25 2020-05-01 5.71 6.44 6.44 6.44 CO DR PEPPER 26 2020-05-08 6.37 1.31 1.31 1.31 CO DR PEPPER 27 2020-05-15 5.92 0.0 0.0 0.0 CO DR PEPPER 28 2020-05-22 7.05 5.47 5.47 5.47 CO DR PEPPER 29 2020-05-29 9.37 23.7 23.7 23.7 CO DR PEPPER 30 2020-06-05 5.02 0.0 0.0 0.0 CO DR PEPPER 31 2020-06-12 7.42 9.26 9.51 9.51 CO DR PEPPER 32 2020-06-19 5.79 27.69 27.92 28.16 CO DR PEPPER 33 2020-06-26 6.48 12.54 12.54 12.78 CO DR PEPPER 34 2020-07-03 5.65 12.45 12.45 12.68 CO DR PEPPER 35 2020-07-10 10.78 10.3 10.3 10.63 CO DR PEPPER 36 2020-07-17 6.8 21.48 21.75 22.01 CO DR PEPPER 37 2020-07-24 7.25 11.89 12.16 12.16 CO DR PEPPER 38 2020-07-31 6.65 1.73 1.73 1.73 CO DR PEPPER 39 2020-08-07 8.17 22.68 22.68 22.97 CO DR PEPPER 40 2020-08-14 8.08 6.55 6.55 6.83 CO DR PEPPER 41 2020-08-21 7.86 28.69 28.69 28.69 CO DR PEPPER 42 2020-08-28 7.35 6.26 6.26 6.26 CO DR PEPPER 43 2020-09-04 5.36 11.21 11.21 11.21 CO DR PEPPER 44 2020-09-11 10.69 18.68 18.68 18.33 CO DR PEPPER 45 2020-09-18 5.33 9.84 9.84 9.62 CO DR PEPPER 46 2020-09-25 4.94 0.0 0.0 0.0 CO DR PEPPER 47 2020-10-02 5.08 10.11 9.88 9.65 CO DR PEPPER 48 2020-10-09 6.7 9.49 8.99 8.74 CO DR PEPPER 49 2020-10-16 5.03 10.82 10.38 9.94 CO DR PEPPER 50 2020-10-23 4.53 22.34 22.12 21.68 CO DR PEPPER 51 2020-10-30 5.24 19.3 19.08 18.4 CO DR PEPPER 52 2020-11-06 6.41 33.14 32.86 32.31 CO DR PEPPER 53 2020-11-13 4.69 30.07 29.85 29.42 CO DR PEPPER 54 2020-11-20 6.04 53.25 52.98 52.45 CO DR PEPPER 55 2020-12-04 6.96 0.74 0.49 0.25 CO DR PEPPER 56 2020-12-11 4.91 47.23 47.46 47.01 CO DR PEPPER 57 2020-12-18 6.02 54.3 54.57 54.3 CO DR PEPPER 58 2020-12-25 7.24 30.73 31.01 31.01 CO DR PEPPER 59 2021-01-01 6.0 25.25 25.5 25.5 CO DR PEPPER 60 2021-01-08 6.77 49.21 49.81 50.12 CO DR PEPPER 61 2021-01-15 4.86 42.95 43.16 43.58 CO DR PEPPER 62 2021-01-22 7.0 44.45 44.74 45.32 CO DR PEPPER 63 2021-01-29 4.98 36.18 36.4 36.84 CO DR PEPPER 64 2021-02-05 5.83 36.38 36.38 36.89 CO DR PEPPER 65 2021-02-12 7.85 25.42 25.42 25.73 CO DR PEPPER 66 2021-02-19 5.39 22.39 22.39 22.39 CO DR PEPPER 67 2021-02-26 4.88 51.5 51.5 51.5 CO DR PEPPER 68 2021-03-05 5.58 1.7 1.7 1.46 CO DR PEPPER 69 2021-03-12 6.31 6.68 6.42 6.42 CO DR PEPPER 70 2021-03-19 6.98 0.0 0.0 0.0 CO DR PEPPER 71 2021-03-26 6.29 6.06 5.82 5.57 CO DR PEPPER 72 2021-04-02 4.94 0.0 0.0 0.0 CO DR PEPPER 73 2021-04-09 8.8 12.92 12.63 12.34 CO DR PEPPER 74 2021-04-16 5.47 4.79 4.79 4.35 CO DR PEPPER 75 2021-04-23 4.88 4.34 4.14 3.94 CO DR PEPPER 76 2021-04-30 5.04 10.68 10.68 10.48 CO DR PEPPER 77 2021-05-07 6.31 12.25 12.25 12.25 CO DR PEPPER 78 2021-05-14 5.72 16.71 16.71 16.71 CO DR PEPPER 79 2021-05-21 5.28 9.89 10.1 10.1 CO DR PEPPER 80 2021-05-28 6.04 8.53 8.75 8.98 CO DR PEPPER 81 2021-06-04 9

Output can only be rendered in Databricks

Output can only be rendered in Databricks

## State + Key Package

## State + Key Packaege

#### -- Physical Volume + Promotion --  
Of the observed instances, the results imply an interesting narrative around the leading package type (**12OZ 12-Pack Cans**).
* This package type has been under extended periods of significant, uninterrupted promotion. This can complicate interpretations.
  * Constant promotion of this package type could influence its ranking among other package types
    * With limited promotion dollars, this is a potential scenario ripe for improvements
      * For example, diverting promotional dollars to **2L Bottles** may yield greater overall volume for a brand
  * The persistent promotion could be a cause and an effect on reduced promotions causing dips in volume
    * There is also possibly a scale factor involved.
    * A metric evaluating volume per promotion units could be valuable.


#### -- Promotion Comparison --  
* The three Dollar Promotion values per period, which are each computed on different average standard price windows, mostly agree.
  * To the extent that they differ, smaller price windows tend to have higher promotion values

In [0]:
p_state = kp_model.states_list[8]
package = kp_model.keys_list[4]
state_table_desc(package, p_state)

display(kp_model.get_filtered_series(p_state, package))

12OZ 12PK CAN in WA


state key_package period_number swire_period_date avg_standard_physical_volume dollar_promo_6wk dollar_promo_8wk dollar_promo_12wk WA 12OZ 12PK CAN 1 2019-11-15 10.43 46.91 46.91 46.91 WA 12OZ 12PK CAN 2 2019-11-22 11.12 51.15 51.15 51.15 WA 12OZ 12PK CAN 3 2019-11-29 20.64 109.01 109.01 109.01 WA 12OZ 12PK CAN 4 2019-12-06 13.98 59.83 59.83 59.83 WA 12OZ 12PK CAN 5 2019-12-13 11.14 51.94 51.94 51.94 WA 12OZ 12PK CAN 6 2019-12-20 11.0 51.92 51.7 51.48 WA 12OZ 12PK CAN 7 2019-12-27 22.67 117.86 117.4 116.95 WA 12OZ 12PK CAN 8 2020-01-03 14.28 67.38 66.81 66.52 WA 12OZ 12PK CAN 9 2020-01-10 11.09 53.23 52.79 52.34 WA 12OZ 12PK CAN 10 2020-01-17 11.6 54.52 54.29 53.59 WA 12OZ 12PK CAN 11 2020-01-24 9.45 40.09 39.9 39.33 WA 12OZ 12PK CAN 12 2020-01-31 8.43 37.74 37.58 37.07 WA 12OZ 12PK CAN 13 2020-02-07 21.24 118.49 117.64 115.94 WA 12OZ 12PK CAN 14 2020-02-14 7.73 32.33 32.02 31.4 WA 12OZ 12PK CAN 15 2020-02-21 12.86 64.58 63.81 62.52 WA 12OZ 12PK CAN 16 2020-02-28 7.97 33.29 32.66 31.7 WA 12OZ 12PK CAN 17 2020-03-06 12.41 56.07 54.33 52.85 WA 12OZ 12PK CAN 18 2020-03-13 9.06 39.7 38.62 37.17 WA 12OZ 12PK CAN 19 2020-03-20 19.29 85.24 83.7 80.23 WA 12OZ 12PK CAN 20 2020-03-27 15.91 64.89 63.94 61.08 WA 12OZ 12PK CAN 21 2020-04-03 10.64 45.31 44.67 42.97 WA 12OZ 12PK CAN 22 2020-04-10 16.08 70.41 69.44 67.19 WA 12OZ 12PK CAN 23 2020-04-17 21.52 118.76 117.9 115.75 WA 12OZ 12PK CAN 24 2020-04-24 10.64 42.79 42.58 41.52 WA 12OZ 12PK CAN 25 2020-05-01 14.28 61.4 61.12 59.98 WA 12OZ 12PK CAN 26 2020-05-08 15.35 69.38 69.08 68.46 WA 12OZ 12PK CAN 27 2020-05-15 14.96 68.52 68.52 67.92 WA 12OZ 12PK CAN 28 2020-05-22 11.76 46.33 46.57 46.57 WA 12OZ 12PK CAN 29 2020-05-29 25.42 138.28 139.3 139.81 WA 12OZ 12PK CAN 30 2020-06-05 10.47 38.32 39.16 39.58 WA 12OZ 12PK CAN 31 2020-06-12 16.09 74.04 75.65 76.93 WA 12OZ 12PK CAN 32 2020-06-19 17.39 75.45 77.54 79.28 WA 12OZ 12PK CAN 33 2020-06-26 16.49 68.95 70.6 72.58 WA 12OZ 12PK CAN 34 2020-07-03 13.08 51.27 51.8 53.89 WA 12OZ 12PK CAN 35 2020-07-10 28.02 133.94 135.06 139.54 WA 12OZ 12PK CAN 36 2020-07-17 15.13 60.82 61.73 63.55 WA 12OZ 12PK CAN 37 2020-07-24 12.23 31.55 32.29 33.75 WA 12OZ 12PK CAN 38 2020-07-31 11.81 26.7 27.65 28.83 WA 12OZ 12PK CAN 39 2020-08-07 24.97 105.37 106.37 108.87 WA 12OZ 12PK CAN 40 2020-08-14 18.08 53.52 53.88 55.32 WA 12OZ 12PK CAN 41 2020-08-21 27.62 104.42 104.97 106.63 WA 12OZ 12PK CAN 42 2020-08-28 16.72 38.79 38.79 39.46 WA 12OZ 12PK CAN 43 2020-09-04 17.96 40.23 39.51 40.23 WA 12OZ 12PK CAN 44 2020-09-11 38.4 178.18 176.64 176.64 WA 12OZ 12PK CAN 45 2020-09-18 21.16 77.43 76.58 76.16 WA 12OZ 12PK CAN 46 2020-09-25 15.86 36.49 36.49 35.85 WA 12OZ 12PK CAN 47 2020-10-02 17.32 55.09 55.44 54.75 WA 12OZ 12PK CAN 48 2020-10-09 24.72 100.38 100.88 99.89 WA 12OZ 12PK CAN 49 2020-10-16 21.5 82.13 81.7 81.27 WA 12OZ 12PK CAN 50 2020-10-23 14.16 33.98 33.7 33.42 WA 12OZ 12PK CAN 51 2020-10-30 15.04 43.62 43.62 43.01 WA 12OZ 12PK CAN 52 2020-11-06 16.98 47.88 47.2 46.86 WA 12OZ 12PK CAN 53 2020-11-13 14.56 34.94 34.36 33.78 WA 12OZ 12PK CAN 54 2020-11-20 17.22 51.66 51.32 50.63 WA 12OZ 12PK CAN 55 2020-12-04 21.81 75.9 75.46 74.59 WA 12OZ 12PK CAN 56 2020-12-11 15.02 33.66 33.66 33.05 WA 12OZ 12PK CAN 57 2020-12-18 17.36 53.12 53.12 52.43 WA 12OZ 12PK CAN 58 2020-12-25 20.79 71.52 71.52 71.1 WA 12OZ 12PK CAN 59 2021-01-01 18.6 55.82 55.82 55.82 WA 12OZ 12PK CAN 60 2021-01-08 16.56 39.76 39.76 39.42 WA 12OZ 12PK CAN 61 2021-01-15 16.31 51.87 51.87 51.87 WA 12OZ 12PK CAN 62 2021-01-22 13.16 31.32 31.32 31.58 WA 12OZ 12PK CAN 63 2021-01-29 12.76 27.83 27.83 28.34 WA 12OZ 12PK CAN 64 2021-02-05 13.44 38.42 38.69 38.96 WA 12OZ 12PK CAN 65 2021-02-12 29.39 125.2 126.38 126.96 WA 12OZ 12PK CAN 66 2021-02-19 11.67 22.63 23.1 23.56 WA 12OZ 12PK CAN 67 2021-02-26 13.03 34.91 34.91 35.43 WA 12OZ 12PK CAN 68 2021-03-05 10.84 22.97 22.54 22.75 WA 12OZ 12PK CAN 69 2021-03-12 13.62 39.51 38.7 38.7 WA 12OZ 12PK CAN 70 2021-03-19 11.01 25.09 24.43 23.99 WA 12OZ 12PK CAN 71 2021-03-

Output can only be rendered in Databricks

Output can only be rendered in Databricks

In [0]:
p_state = kp_model.states_list[8]
package = kp_model.keys_list[1]
state_table_desc(package, p_state)

display(kp_model.get_filtered_series(p_state, package))

.5L 6PK BOTTLE in WA


state key_package period_number swire_period_date avg_standard_physical_volume dollar_promo_6wk dollar_promo_8wk dollar_promo_12wk WA .5L 6PK BOTTLE 1 2019-11-15 1.51 2.84 2.84 2.84 WA .5L 6PK BOTTLE 2 2019-11-22 1.43 2.86 2.86 2.86 WA .5L 6PK BOTTLE 3 2019-11-29 1.55 2.97 2.97 2.97 WA .5L 6PK BOTTLE 4 2019-12-06 1.52 2.79 2.79 2.79 WA .5L 6PK BOTTLE 5 2019-12-13 2.53 18.74 18.74 18.74 WA .5L 6PK BOTTLE 6 2019-12-20 1.48 3.2 3.08 3.02 WA .5L 6PK BOTTLE 7 2019-12-27 1.74 4.09 3.89 3.68 WA .5L 6PK BOTTLE 8 2020-01-03 1.57 4.39 4.08 3.76 WA .5L 6PK BOTTLE 9 2020-01-10 1.88 11.04 10.59 10.06 WA .5L 6PK BOTTLE 10 2020-01-17 1.52 5.12 4.64 4.15 WA .5L 6PK BOTTLE 11 2020-01-24 1.79 10.71 10.07 9.35 WA .5L 6PK BOTTLE 12 2020-01-31 1.3 4.44 4.12 3.5 WA .5L 6PK BOTTLE 13 2020-02-07 2.2 15.49 15.4 14.26 WA .5L 6PK BOTTLE 14 2020-02-14 2.22 14.76 14.67 13.69 WA .5L 6PK BOTTLE 15 2020-02-21 2.45 18.15 18.15 17.56 WA .5L 6PK BOTTLE 16 2020-02-28 2.41 17.14 17.24 16.95 WA .5L 6PK BOTTLE 17 2020-03-06 1.5 4.7 4.82 4.82 WA .5L 6PK BOTTLE 18 2020-03-13 2.45 14.77 15.06 15.26 WA .5L 6PK BOTTLE 19 2020-03-20 2.07 5.95 6.36 6.61 WA .5L 6PK BOTTLE 20 2020-03-27 2.33 10.08 10.54 11.01 WA .5L 6PK BOTTLE 21 2020-04-03 2.73 18.05 18.38 19.04 WA .5L 6PK BOTTLE 22 2020-04-10 1.81 4.57 4.64 5.15 WA .5L 6PK BOTTLE 23 2020-04-17 1.83 4.48 4.55 4.92 WA .5L 6PK BOTTLE 24 2020-04-24 2.24 8.78 8.87 9.23 WA .5L 6PK BOTTLE 25 2020-05-01 1.86 4.69 4.69 4.92 WA .5L 6PK BOTTLE 26 2020-05-08 2.01 6.91 6.91 6.99 WA .5L 6PK BOTTLE 27 2020-05-15 2.26 9.47 9.47 9.65 WA .5L 6PK BOTTLE 28 2020-05-22 1.85 5.4 5.55 5.62 WA .5L 6PK BOTTLE 29 2020-05-29 1.75 4.77 4.91 4.98 WA .5L 6PK BOTTLE 30 2020-06-05 1.89 6.89 7.04 7.12 WA .5L 6PK BOTTLE 31 2020-06-12 2.26 10.5 10.59 10.77 WA .5L 6PK BOTTLE 32 2020-06-19 1.6 2.37 2.5 2.63 WA .5L 6PK BOTTLE 33 2020-06-26 1.7 3.46 3.53 3.67 WA .5L 6PK BOTTLE 34 2020-07-03 2.45 13.13 13.13 13.33 WA .5L 6PK BOTTLE 35 2020-07-10 2.38 10.97 10.97 11.07 WA .5L 6PK BOTTLE 36 2020-07-17 2.06 4.04 4.04 4.13 WA .5L 6PK BOTTLE 37 2020-07-24 3.77 19.78 19.63 19.63 WA .5L 6PK BOTTLE 38 2020-07-31 2.97 10.34 10.22 10.1 WA .5L 6PK BOTTLE 39 2020-08-07 3.33 16.5 16.37 16.11 WA .5L 6PK BOTTLE 40 2020-08-14 2.68 10.83 10.61 10.4 WA .5L 6PK BOTTLE 41 2020-08-21 2.22 4.88 4.62 4.44 WA .5L 6PK BOTTLE 42 2020-08-28 5.74 44.73 43.82 43.13 WA .5L 6PK BOTTLE 43 2020-09-04 3.03 12.99 12.63 12.02 WA .5L 6PK BOTTLE 44 2020-09-11 3.25 14.96 14.57 13.92 WA .5L 6PK BOTTLE 45 2020-09-18 2.51 6.51 6.41 5.81 WA .5L 6PK BOTTLE 46 2020-09-25 2.57 8.12 8.12 7.5 WA .5L 6PK BOTTLE 47 2020-10-02 2.7 10.28 10.28 9.95 WA .5L 6PK BOTTLE 48 2020-10-09 4.18 25.23 25.07 24.73 WA .5L 6PK BOTTLE 49 2020-10-16 3.81 19.05 19.05 18.75 WA .5L 6PK BOTTLE 50 2020-10-23 5.62 30.8 30.8 30.57 WA .5L 6PK BOTTLE 51 2020-10-30 5.02 31.93 31.73 31.53 WA .5L 6PK BOTTLE 52 2020-11-06 3.45 18.74 18.74 18.47 WA .5L 6PK BOTTLE 53 2020-11-13 7.66 61.87 61.26 60.65 WA .5L 6PK BOTTLE 54 2020-11-20 5.38 32.73 32.3 31.86 WA .5L 6PK BOTTLE 55 2020-12-04 3.04 12.28 12.04 11.67 WA .5L 6PK BOTTLE 56 2020-12-11 6.72 54.32 53.78 52.97 WA .5L 6PK BOTTLE 57 2020-12-18 3.12 14.85 14.35 13.98 WA .5L 6PK BOTTLE 58 2020-12-25 4.98 33.05 32.65 31.66 WA .5L 6PK BOTTLE 59 2021-01-01 3.72 18.45 18.3 17.71 WA .5L 6PK BOTTLE 60 2021-01-08 6.86 57.35 57.08 56.25 WA .5L 6PK BOTTLE 61 2021-01-15 2.92 14.02 13.9 13.67 WA .5L 6PK BOTTLE 62 2021-01-22 3.39 12.36 12.36 12.22 WA .5L 6PK BOTTLE 63 2021-01-29 6.07 43.44 43.93 43.93 WA .5L 6PK BOTTLE 64 2021-02-05 6.46 40.34 41.12 41.63 WA .5L 6PK BOTTLE 65 2021-02-12 4.83 27.79 28.18 28.76 WA .5L 6PK BOTTLE 66 2021-02-19 5.16 31.34 31.75 32.37 WA .5L 6PK BOTTLE 67 2021-02-26 3.41 17.88 18.02 18.56 WA .5L 6PK BOTTLE 68 2021-03-05 3.25 11.85 11.72 12.11 WA .5L 6PK BOTTLE 69 2021-03-12 4.93 29.19 28.99 29.19 WA .5L 6PK BOTTLE 70 2021-03-19 7.66 51.15 50.54 50.54 WA .5L 6PK BOTTLE 71 2021-03-26 3.93 20.44 20.28 19.96 WA .5L 6PK BOTTLE 72 2021-04-02 3.92 16.93 16.62 16.31 WA .5L 6PK BOTTLE 73

Output can only be rendered in Databricks

Output can only be rendered in Databricks

In [0]:
p_state = kp_model.states_list[7]
package = kp_model.keys_list[4]
state_table_desc(package, p_state)

display(kp_model.get_filtered_series(p_state, package))

12OZ 12PK CAN in OR


state key_package period_number swire_period_date avg_standard_physical_volume dollar_promo_6wk dollar_promo_8wk dollar_promo_12wk OR 12OZ 12PK CAN 1 2019-11-15 7.49 28.89 28.89 28.89 OR 12OZ 12PK CAN 2 2019-11-22 10.03 49.52 49.52 49.52 OR 12OZ 12PK CAN 3 2019-11-29 19.92 113.17 113.17 113.17 OR 12OZ 12PK CAN 4 2019-12-06 10.65 50.06 50.06 50.06 OR 12OZ 12PK CAN 5 2019-12-13 11.37 55.06 55.06 55.06 OR 12OZ 12PK CAN 6 2019-12-20 10.54 50.83 50.83 50.62 OR 12OZ 12PK CAN 7 2019-12-27 21.4 119.41 119.41 118.98 OR 12OZ 12PK CAN 8 2020-01-03 12.44 63.72 63.47 63.22 OR 12OZ 12PK CAN 9 2020-01-10 10.37 53.09 52.68 52.47 OR 12OZ 12PK CAN 10 2020-01-17 7.31 28.38 28.24 27.94 OR 12OZ 12PK CAN 11 2020-01-24 5.84 25.0 24.88 24.64 OR 12OZ 12PK CAN 12 2020-01-31 10.18 52.33 52.12 51.71 OR 12OZ 12PK CAN 13 2020-02-07 21.62 127.13 126.69 125.4 OR 12OZ 12PK CAN 14 2020-02-14 6.35 28.58 28.45 28.19 OR 12OZ 12PK CAN 15 2020-02-21 8.59 39.37 39.02 38.51 OR 12OZ 12PK CAN 16 2020-02-28 6.53 26.9 26.51 26.12 OR 12OZ 12PK CAN 17 2020-03-06 10.56 49.42 48.36 47.52 OR 12OZ 12PK CAN 18 2020-03-13 7.79 33.81 33.03 32.25 OR 12OZ 12PK CAN 19 2020-03-20 15.83 72.5 71.24 69.02 OR 12OZ 12PK CAN 20 2020-03-27 12.58 49.06 47.8 45.79 OR 12OZ 12PK CAN 21 2020-04-03 7.59 31.9 31.14 29.77 OR 12OZ 12PK CAN 22 2020-04-10 9.08 30.53 29.62 27.8 OR 12OZ 12PK CAN 23 2020-04-17 14.46 72.01 70.56 67.96 OR 12OZ 12PK CAN 24 2020-04-24 7.99 27.65 27.17 25.73 OR 12OZ 12PK CAN 25 2020-05-01 12.18 55.81 55.08 52.88 OR 12OZ 12PK CAN 26 2020-05-08 14.44 67.87 67.0 64.98 OR 12OZ 12PK CAN 27 2020-05-15 8.9 34.87 34.51 33.62 OR 12OZ 12PK CAN 28 2020-05-22 13.08 62.5 62.5 61.45 OR 12OZ 12PK CAN 29 2020-05-29 25.13 144.27 145.78 145.28 OR 12OZ 12PK CAN 30 2020-06-05 8.55 23.43 24.28 24.62 OR 12OZ 12PK CAN 31 2020-06-12 15.79 65.05 67.27 68.84 OR 12OZ 12PK CAN 32 2020-06-19 19.89 94.68 97.86 101.04 OR 12OZ 12PK CAN 33 2020-06-26 12.36 42.25 44.23 46.7 OR 12OZ 12PK CAN 34 2020-07-03 11.0 34.78 36.1 38.52 OR 12OZ 12PK CAN 35 2020-07-10 33.57 173.87 176.55 184.61 OR 12OZ 12PK CAN 36 2020-07-17 13.15 46.53 46.53 49.43 OR 12OZ 12PK CAN 37 2020-07-24 15.54 59.05 59.05 61.54 OR 12OZ 12PK CAN 38 2020-07-31 10.63 26.15 26.58 27.85 OR 12OZ 12PK CAN 39 2020-08-07 21.33 86.17 87.45 89.59 OR 12OZ 12PK CAN 40 2020-08-14 15.66 36.96 37.9 38.52 OR 12OZ 12PK CAN 41 2020-08-21 22.17 58.99 60.32 60.32 OR 12OZ 12PK CAN 42 2020-08-28 15.82 37.02 37.34 37.97 OR 12OZ 12PK CAN 43 2020-09-04 20.42 57.99 58.4 59.63 OR 12OZ 12PK CAN 44 2020-09-11 54.19 265.51 265.51 268.76 OR 12OZ 12PK CAN 45 2020-09-18 15.18 32.17 31.87 32.78 OR 12OZ 12PK CAN 46 2020-09-25 18.05 49.47 49.47 50.19 OR 12OZ 12PK CAN 47 2020-10-02 16.07 48.87 48.87 48.87 OR 12OZ 12PK CAN 48 2020-10-09 32.16 153.11 153.75 153.75 OR 12OZ 12PK CAN 49 2020-10-16 25.09 121.46 121.46 121.46 OR 12OZ 12PK CAN 50 2020-10-23 15.54 49.11 48.8 49.11 OR 12OZ 12PK CAN 51 2020-10-30 14.86 41.32 41.03 41.03 OR 12OZ 12PK CAN 52 2020-11-06 17.77 46.91 46.56 46.56 OR 12OZ 12PK CAN 53 2020-11-13 17.89 50.45 50.09 49.73 OR 12OZ 12PK CAN 54 2020-11-20 17.22 47.18 46.84 46.15 OR 12OZ 12PK CAN 55 2020-12-04 21.49 81.66 81.66 81.23 OR 12OZ 12PK CAN 56 2020-12-11 14.19 34.06 34.06 34.06 OR 12OZ 12PK CAN 57 2020-12-18 14.27 32.25 32.25 32.54 OR 12OZ 12PK CAN 58 2020-12-25 18.17 53.04 53.04 53.77 OR 12OZ 12PK CAN 59 2021-01-01 16.7 41.74 42.41 43.07 OR 12OZ 12PK CAN 60 2021-01-08 19.95 56.26 57.46 57.86 OR 12OZ 12PK CAN 61 2021-01-15 17.12 47.61 48.29 49.32 OR 12OZ 12PK CAN 62 2021-01-22 12.53 28.57 28.82 29.57 OR 12OZ 12PK CAN 63 2021-01-29 14.72 24.15 23.85 25.03 OR 12OZ 12PK CAN 64 2021-02-05 11.31 14.7 14.24 14.47 OR 12OZ 12PK CAN 65 2021-02-12 37.9 107.65 103.86 100.83 OR 12OZ 12PK CAN 66 2021-02-19 11.18 20.36 19.24 17.67 OR 12OZ 12PK CAN 67 2021-02-26 13.11 35.67 33.84 31.74 OR 12OZ 12PK CAN 68 2021-03-05 10.29 24.89 23.04 20.98 OR 12OZ 12PK CAN 69 2021-03-12 14.48 42.59 39.69 37.08 OR 12OZ 12PK CAN 70 2021-03-19 10.22 24.11 22.88 20.84 OR 12OZ 12PK CAN 71 2021-03-26 12

Output can only be rendered in Databricks

Output can only be rendered in Databricks

In [0]:
p_state = kp_model.states_list[7]
package = kp_model.keys_list[20]
state_table_desc(package, p_state)

display(kp_model.get_filtered_series(p_state, package))

2L SINGLE BOTTLE in OR


state key_package period_number swire_period_date avg_standard_physical_volume dollar_promo_6wk dollar_promo_8wk dollar_promo_12wk OR 2L SINGLE BOTTLE 1 2019-11-15 3.26 6.78 6.78 6.78 OR 2L SINGLE BOTTLE 2 2019-11-22 4.58 15.37 15.37 15.37 OR 2L SINGLE BOTTLE 3 2019-11-29 4.78 21.79 21.79 21.79 OR 2L SINGLE BOTTLE 4 2019-12-06 4.12 10.22 10.22 10.22 OR 2L SINGLE BOTTLE 5 2019-12-13 4.88 23.02 23.02 23.02 OR 2L SINGLE BOTTLE 6 2019-12-20 2.85 3.64 3.64 3.64 OR 2L SINGLE BOTTLE 7 2019-12-27 4.47 17.88 17.88 17.88 OR 2L SINGLE BOTTLE 8 2020-01-03 4.91 12.96 12.96 12.96 OR 2L SINGLE BOTTLE 9 2020-01-10 2.93 3.29 3.29 3.29 OR 2L SINGLE BOTTLE 10 2020-01-17 4.11 11.85 11.85 11.85 OR 2L SINGLE BOTTLE 11 2020-01-24 5.49 26.36 26.36 26.36 OR 2L SINGLE BOTTLE 12 2020-01-31 3.47 14.72 14.72 14.72 OR 2L SINGLE BOTTLE 13 2020-02-07 3.22 1.54 1.29 1.29 OR 2L SINGLE BOTTLE 14 2020-02-14 4.7 19.17 19.17 18.8 OR 2L SINGLE BOTTLE 15 2020-02-21 3.51 15.44 15.16 15.16 OR 2L SINGLE BOTTLE 16 2020-02-28 2.9 1.39 1.39 1.16 OR 2L SINGLE BOTTLE 17 2020-03-06 3.71 5.94 5.64 5.34 OR 2L SINGLE BOTTLE 18 2020-03-13 5.1 17.15 17.15 16.74 OR 2L SINGLE BOTTLE 19 2020-03-20 3.72 2.38 2.38 2.08 OR 2L SINGLE BOTTLE 20 2020-03-27 4.47 18.58 18.58 18.23 OR 2L SINGLE BOTTLE 21 2020-04-03 5.79 28.71 28.71 28.24 OR 2L SINGLE BOTTLE 22 2020-04-10 3.65 2.04 2.04 2.04 OR 2L SINGLE BOTTLE 23 2020-04-17 4.35 6.95 6.95 6.95 OR 2L SINGLE BOTTLE 24 2020-04-24 3.46 2.21 2.21 2.21 OR 2L SINGLE BOTTLE 25 2020-05-01 3.8 6.99 6.99 6.99 OR 2L SINGLE BOTTLE 26 2020-05-08 5.4 15.98 15.98 15.98 OR 2L SINGLE BOTTLE 27 2020-05-15 4.96 19.44 19.44 19.84 OR 2L SINGLE BOTTLE 28 2020-05-22 5.25 18.08 18.08 18.5 OR 2L SINGLE BOTTLE 29 2020-05-29 5.51 10.14 10.14 10.14 OR 2L SINGLE BOTTLE 30 2020-06-05 3.72 1.78 1.78 1.78 OR 2L SINGLE BOTTLE 31 2020-06-12 4.01 2.25 2.25 2.25 OR 2L SINGLE BOTTLE 32 2020-06-19 4.14 7.29 7.29 7.29 OR 2L SINGLE BOTTLE 33 2020-06-26 5.88 16.93 16.93 16.93 OR 2L SINGLE BOTTLE 34 2020-07-03 8.05 36.05 36.05 36.05 OR 2L SINGLE BOTTLE 35 2020-07-10 7.45 29.81 29.81 29.81 OR 2L SINGLE BOTTLE 36 2020-07-17 5.13 9.02 9.02 9.02 OR 2L SINGLE BOTTLE 37 2020-07-24 3.94 6.62 6.31 6.31 OR 2L SINGLE BOTTLE 38 2020-07-31 3.89 3.74 3.42 3.42 OR 2L SINGLE BOTTLE 39 2020-08-07 4.69 8.25 7.87 7.87 OR 2L SINGLE BOTTLE 40 2020-08-14 4.25 9.18 9.18 8.84 OR 2L SINGLE BOTTLE 41 2020-08-21 4.81 19.62 19.62 19.62 OR 2L SINGLE BOTTLE 42 2020-08-28 3.71 1.78 1.78 1.78 OR 2L SINGLE BOTTLE 43 2020-09-04 8.73 49.56 49.56 49.56 OR 2L SINGLE BOTTLE 44 2020-09-11 4.02 1.93 1.93 1.93 OR 2L SINGLE BOTTLE 45 2020-09-18 4.49 16.52 16.52 16.52 OR 2L SINGLE BOTTLE 46 2020-09-25 3.72 2.08 1.79 2.08 OR 2L SINGLE BOTTLE 47 2020-10-02 7.14 35.42 35.42 35.42 OR 2L SINGLE BOTTLE 48 2020-10-09 3.83 4.6 4.6 4.6 OR 2L SINGLE BOTTLE 49 2020-10-16 4.85 13.97 13.97 13.97 OR 2L SINGLE BOTTLE 50 2020-10-23 4.56 9.12 9.12 9.12 OR 2L SINGLE BOTTLE 51 2020-10-30 4.51 18.78 18.78 18.78 OR 2L SINGLE BOTTLE 52 2020-11-06 6.38 21.43 21.43 21.43 OR 2L SINGLE BOTTLE 53 2020-11-13 3.83 4.29 4.29 4.29 OR 2L SINGLE BOTTLE 54 2020-11-20 6.28 30.63 30.63 30.63 OR 2L SINGLE BOTTLE 55 2020-12-04 5.04 8.87 8.87 8.87 OR 2L SINGLE BOTTLE 56 2020-12-11 4.4 8.09 8.09 8.44 OR 2L SINGLE BOTTLE 57 2020-12-18 4.53 11.59 11.59 11.59 OR 2L SINGLE BOTTLE 58 2020-12-25 6.99 24.05 24.6 24.6 OR 2L SINGLE BOTTLE 59 2021-01-01 7.12 42.71 43.27 43.84 OR 2L SINGLE BOTTLE 60 2021-01-08 7.05 21.45 22.01 22.01 OR 2L SINGLE BOTTLE 61 2021-01-15 3.55 3.98 4.26 4.55 OR 2L SINGLE BOTTLE 62 2021-01-22 6.21 27.82 28.32 28.81 OR 2L SINGLE BOTTLE 63 2021-01-29 6.01 26.43 26.43 26.91 OR 2L SINGLE BOTTLE 64 2021-02-05 6.65 28.72 28.72 29.25 OR 2L SINGLE BOTTLE 65 2021-02-12 5.86 16.86 16.86 16.86 OR 2L SINGLE BOTTLE 66 2021-02-19 5.66 23.54 23.54 23.54 OR 2L SINGLE BOTTLE 67 2021-02-26 5.59 22.81 22.81 22.81 OR 2L SINGLE BOTTLE 68 2021-03-05 5.01 20.05 20.05 20.05 OR 2L SINGLE BOTTLE 69 2021-03-12 4.81 14.62 14.62 14.62 OR 2L SINGLE BOTTLE 70 2021-03-19 3.38 12.

Output can only be rendered in Databricks

Output can only be rendered in Databricks